packages/core-data/src/utils/log-entity-deprecation.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * WordPress dependencies
+ */
+import deprecated from '@wordpress/deprecated';
+
+/**
+ * Internal dependencies
+ */
+import { deprecatedEntities } from '../entities';
+
+let loggedAlready = false;
+
+/**
+ * Logs a deprecation warning for an entity, if it's deprecated.
+ *
+ * @param kind                    - The kind of the entity.
+ * @param name                    - The name of the entity.
+ * @param functionName            - The name of the function that was called with a deprecated entity.
+ * @param alternativeFunctionName - The name of the alternative function that should be used instead.
+ */
+export default function logEntityDeprecation(
+	kind: string,
+	name: string,
+	functionName: string,
+	alternativeFunctionName?: string
+) {
+	const deprecation = deprecatedEntities[ kind ]?.[ name ];
+	if ( ! deprecation ) {
+		return;
+	}
+
+	const { alternative } = deprecation;
+	let alternativeMessage = `The '${ alternative.kind }', '${ alternative.name }' entity`;
```

Comment:
Here are some messages I'm seeing in the console:

`The 'root', 'media' entity (used via 'getMediaItems') is deprecated since version 6.9. Please use The 'postType', 'attachment' entity via the 'getEntityRecords' function instead.`

It's a teensy bit misleading as `getMediaItems` doesn't accept kind/name args.

`The 'root', 'media' entity (used via 'getEntityRecord') is deprecated since version 6.9. Please use The 'postType', 'attachment' entity instead.`

Would it be clearer if we specified that the 'root', 'media' _arguments_ are deprecated?

Also , `"The"` has a capital T. 

No kittens will die if the console message stays as you have it.

Reply:
This should be solved now, it'll log something like this:
```
getMediaItems is deprecated since version 6.9. Please use the 'postType', 'attachment' entity via the 'getEntityRecords' function instead.
```

Reply:
The only thing I'm perhaps mixed on is whether we should say `'getEntityRecords' selector` instead of function. It means adding more configuration everywhere, so that's why I'm avoiding it.